### PR TITLE
Apply PF Breaking Change Best Practices to MUI Theme Stylesheet

### PR DESCRIPTION
**Pull request by jenny-s51**
## Description

Updates theming overrides to align with [PF breaking change standards](https://docs.google.com/document/d/1HipMvYqlUpTIP2H4QtCMAGdsu8TCCW7XKuR4imGarkA/edit#heading=h.c9wk66om5xcd).

## How Has This Been Tested?

N/A, refactoring only. 
To test modal refactoring updates: 

1. Model Registry list view -> Actions menu on table row -> "Archive model"
2. Model Registry list view -> Actions menu next to "Register Model" button in toolbar -> Click "View Archived Models" -> Archived models list view -> Actions menu on table row -> "Restore Model"
3. Model Registry list view -> Click on model -> Versions Tab -> Actions menu on table row -> Click "Archive model version"
4. Model Registry list view -> Click on model -> Versions Tab -> Actions menu next to "Register Model" button in toolbar -> Click "View Archived Versions"-> Archived versions list view -> Actions menu on table row -> "Restore Version"

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [x] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

If you have UI changes

- [x] The developer has added tests or explained why testing cannot be added.
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: clients/ui/frontend/src/style/MUI-theme.scss [hunk 1]
```diff
@@ -237,21 +242,24 @@
 }
 
 .pf-v6-c-form-control input {
+  // TODO: https://issues.redhat.com/browse/RHOAIENG-14702,
```
Comment (alexcreasy) [inline clients/ui/frontend/src/style/MUI-theme.scss:245]: Small nit, we shouldn't check Red Hat references into the community codebase, could you remove the link to the internal JIRA please? :)

File: clients/ui/frontend/src/style/MUI-theme.scss [hunk 2]
```diff
@@ -397,6 +405,9 @@
 }
 
 .pf-v6-c-form__group:focus-within .pf-v6-c-form__label {
+  // TODO: https://issues.redhat.com/browse/RHOAIENG-14702
```
Comment (alexcreasy) [inline clients/ui/frontend/src/style/MUI-theme.scss:408]: Likewise here

**Review by alexcreasy — CHANGES_REQUESTED** [review]
Thanks for the contribution Jenny! I haven't tested this out yet, just read through the code, but it looks good. 

There are just a few tiny changes related to checking in Red Hat references to the community codebase to take care of.

I'll update this if I find anything during testing :)

**Review by jenny-s51 — COMMENTED** [review]
Thank you for reviewing @alexcreasy - updated the stylesheet to apply your feedback.

**Comment by lucferbux** [comment]
@jenny-s51 I was trying to spot which css is overriding the button corner radious, but running it i've spotted this issue:
Screenshot 2024-10-18 at 17 03 46
I'm sorry i cannot add it in the right line 😅

File: clients/ui/frontend/src/style/MUI-theme.scss [hunk 3]
```diff
@@ -447,15 +460,19 @@
 }
 
 .pf-v6-c-menu-toggle.pf-m-plain {
+  // Needs clarity --pf-v6-c-menu-toggle--m-plain--BorderRadius does not exist
```
Comment (lucferbux) [inline clients/ui/frontend/src/style/MUI-theme.scss:463]: Ah, this is the right line, but it doesn't seem a new addition, regardless of that, a 50% would make that weird shape in the button

**Comment by jenny-s51** [comment]
Nice catch, thanks @lucferbux ! I've pushed a fix for the buttons and am still investigating as the modals will require some refactoring make sure the error message in the modal is themed correctly. `DashboardModalFooter` currently uses `Stack` and`ActionsList` around the buttons to render the Alert component like this:

Screenshot 2024-10-18 at 5 03 01 PM

When using a form input in a modal like this, the PF recommendation is to render Alerts [at the top of the form](https://staging-v6.patternfly.org/components/forms/form#invalid-with-form-alert). Currently working on updating the modals to apply this change.

**Review by jenny-s51 — COMMENTED** [review]
Updated and refactored the modals to apply the correct styling. The action buttons now have the correct button corner radius and the alert is shown at the top of the modal per the PF design recommendations cc @lucferbux 

Screenshot 2024-10-24 at 1 27 23 PM

**Review by lucferbux — APPROVED** [review]
/lgtm

So far breaking down some components into constants is a really great pattern, we are breaking a little bit the 1:1 match with downstream but checking the latest changes is not that big of a deal and would benefit in the future.

@ederign @alexcreasy could you guys take a look and approve?

**Review by ederign — APPROVED** [review]

**Pull request merged**